### PR TITLE
Handle no nodes and/or no edges in new StellarGraph

### PR DESCRIPTION
**Pull request by huonw**
If there's no node types or no edge types, the code ends up calling `pd.concat([])` which is an error, so this has to be made conditional and appropriately-shaped dataframe created manually.

Now that it works, this also switches the default empty `StellarGraph()` constructor call to use the new `StellarGraph` implementation.

See: #774

### CODE REVIEW

File: stellargraph/core/element_data.py [hunk 1]
```diff
@@ -107,6 +107,8 @@ class ElementData:
         shared (dict of type name to pandas DataFrame): information for the elements of each type
     """
 
+    _SHARED_REQUIRED_COLUMNS = []
```
Comment (timpitman) [inline stellargraph/core/element_data.py]: I know this is called "shared" but possibly an extra comment to say that it's sometimes overridden in derived classes?

**Comment by kieranricardo** [comment]
I think you meant "Handle no nodes and/or no edges in new StellarGraph" in the title? :)

**Comment by huonw** [comment]
This was closed accidentally, see: https://github.com/stellargraph/stellargraph/issues/786.

**Comment by huonw** [comment]
New PR: https://github.com/stellargraph/stellargraph/pull/790

**Pull request closed without merging**